src/app.js

```diff
@@ -1,21 +1,52 @@
+/* eslint-disable no-console */
 'use strict';
 
-/**
- * Implement sum function:
- *
- * Function takes 2 numbers and returns their sum
- *
- * sum(1, 2) === 3
- * sum(1, 11) === 12
- *
- * @param {number} a
- * @param {number} b
- *
- * @return {number}
- */
-function sum(a, b) {
-  // write code here
-  return a + b;
-}
-
-module.exports = sum;
+const http = require('http');
+const fs = require('fs');
+
+const PORT = process.env.PORT || 3001;
+
+const server = http.createServer((req, res) => {
+  const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+
+  const validPath = normalizedUrl.pathname.split('/')[1] === 'file';
+
+  if (!validPath) {
+    res.statusCode = 400;
+    res.end('Path is not correct. Should be /file/<path>');
+  } else {
+    const fileName
+      = normalizedUrl.pathname.split('/').slice(2).join('/') || 'index.html';
+
+    const fileExtension = fileName.split('.').slice(-1);
+
+    const mimeType = {
+      html: 'text/html',
+      css: 'text/css',
+      js: 'text/javascript',
+      jpg: 'image/jpeg',
+      png: 'image/png',
+      json: 'application/json',
+      pdf: 'application/pdf',
+      mp4: 'video/mp4',
+      mp3: 'audio/mpeg',
+    };
+
+    fs.readFile(`./public/${fileName}`, (error, data) => {
```

Comment:
Using a relative path (./public/${pathToFile}) can expose files outside of the intended directory if the request contains '../' in the pathname. To resolve this issue, use the 'path' module to resolve the absolute path and make sure it still resides within the 'public' directory.